soc/arm/cypress/common/cypress_psoc6_dt.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2020 Linaro Ltd.
+ * Copyright (c) 2020 ATL Electronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief Cypress PSoC-6 MCU family devicetree helper macros
+ */
+
+#ifndef _CYPRESS_PSOC6_DT_H_
+#define _CYPRESS_PSOC6_DT_H_
+
+#include <devicetree.h>
+
+/*
+ * Devicetree macros related to interrupt
+ *
+ * The main "API" macro is CY_PSOC6_IRQ_CONFIG. It is an internal definition
+ * used to configure the PSoC-6 interrupts in a generic way. This is necessary
+ * because Cortex-M0+ can handle a limited number of interrupts and have
+ * multiplexers in front of any NVIC interrupt line.
+ *
+ * The CY_PSOC6_IRQ_CONFIG expands from CY_PSOC6_DT_NVIC_INSTALL, the public
+ * API used by drivers. See below code fragment:
+ *
+ * static void <driver>_psoc6_isr(const struct device *dev)
+ * {
+ *     ...
+ * }
+ *
+ * #define <DRIVER>_PSOC6_INIT(n)					\
+ *    ...								\
+ * static void <driver>_psoc6_irq_config(const struct device *port)	\
+ * {									\
+ *      CY_PSOC6_DT_NVIC_INSTALL(n,					\
+ *                               <driver>_psoc6_isr);			\
+ *      };								\
+ * };
+ *
+ * where:
+ *   n   - driver instance number
+ *   isr - isr function to be called
+ *
+ * Cortex-M4 simple pass the parameter and constructs an usual NVIC
+ * configuration code.
+ *
+ * The Cortex-M0+ must get from interrupt parent the interrupt line and
+ * configure the interrupt channel to connect PSoC-6 peripheral interrupt to
+ * Cortex-M0+ NVIC. The multiplexer is configured by CY_PSOC6_DT_NVIC_MUX_MAP
+ * using the interrupt value from the interrupt parent.
+ *
+ * see cypress,psoc6-int-mux.yaml for devicetree documentation.
+ */
+#ifdef CONFIG_CPU_CORTEX_M0PLUS
+/* Cortex-M0+
+ * - install config only when exists an interrupt_parent property
+ * - get peripheral irq using PROP_BY_INDEX, to avoid translation from
+ *   interrupt-parent node property.
+ * - configure interrupt channel using the channel number register value from
+ *   interrupt-parent node.
+ */
+#define CY_PSOC6_DT_NVIC_INSTALL(n, isr)	              \
```

Comment:
Based on what we are doing here, we should do `_DT_INST_`.

Reply:
Fixed!